Cython/Compiler/Parsing.py

```diff
@@ -1143,11 +1147,17 @@ def p_f_string_expr(s, unicode_value, pos, starting_index, is_raw):
         elif c == '#':
             error(_f_string_error_pos(pos, unicode_value, i),
                   "format string cannot include #")
-        elif nested_depth == 0 and c in '!:}':
-            # allow != as a special case
-            if c == '!' and i + 1 < size and unicode_value[i + 1] == '=':
-                i += 1
-                continue
+        elif nested_depth == 0 and c in '><=!:}':
+            # allow special cases with '!' and '='
+            if i + 1 < size:
+                chars = c + unicode_value[i + 1]
+                if chars in ('!=', '==', '>=', '<='):
+                    i += 2  # we checked 2, so we can skip 2
+                    continue
+
+                if c in '><':  # allow single '<' and '>'
+                    i += 1
+                    continue
```

Comment:
We can avoid the string building here.
```suggestion
                if c in '!=><':
                    if unicode_value[i + 1] == '=':
                        i += 2  # we checked 2, so we can skip 2
                        continue
                    elif c in '><':  # allow single '<' and '>'
                        i += 1
                        continue
```